components/ui/FlashCards/MultipleChoice.tsx

```diff
@@ -0,0 +1,206 @@
+import { Text, View, StyleSheet, Pressable } from 'react-native';
+import React, { useState, useMemo } from 'react';
+import { Link } from 'expo-router';
+import { faker } from '@faker-js/faker';
+import MainButton from '../Buttons/MainButton';
+import FlipCard from 'react-native-flip-card';
+
+const initialColorState: Record<string, string> = {
```

Comment:
Sorry I'm gonna be annoying about types. Instead of annotating initialColorState as Record<string, string>, we actually wouldn't want to annotate it at all. 
useState should implicitly know the type once you put the object in. If you want to annotate useState you can do 
`useState<typeof initialColorState>(initialState);`
I made an example you can mess with to see the difference https://www.typescriptlang.org/play?#code/MYewdgzgLgBAlmOUCGAbAyiqBTGBeGAbwCgYyYBBALhgCJaAaU8gIRvqfJgGF3HmyAET5MAvgG5ixKAE8ADrm4hUIAE6ZkOfDFkKQAM3iIUGLNknFQkWAiRw0GnBQgAlbKFUATGm4+eAPNCqCADmDDBBoQB82iRc1HT8XGyJnOS8qQIwwonEolK2UPammtjOvmqeAHQAWtq0EAAWIACuqJ4wYCCwjchyCmC0FgD0wzAAAlAQALTYAB4KwFCzqqpqxIXFjti19cDIYJ3dML392IPiQA

Main reason being if you use Record<string, string>, you're saying this can have any key that is a string, which is not what we want.

Reply:
I kept getting a type error on the handleSelectAnswer answer function without it but I changed initialColorState to:

const initialColorState = {
  A: 'white',  // instead of ' '
  B: 'white',
  C: 'white',
  D: 'white',
};

Then I simplified  handleSelectAnswer. This is mostly due to me not knowing typescript and piling on more typescript when I get typescript errors.